src/ui.rs

```diff
@@ -137,7 +144,7 @@ impl Ui {
     /// Whether continuous feedback should be displayed for long-running
     /// operations
     pub fn use_progress_indicator(&self) -> bool {
-        self.settings().use_progress_indicator() && atty::is(Stream::Stdout)
+        progress_indicator_setting(self.settings()) && atty::is(Stream::Stdout)
```

Comment:
Nit: Instead of passing in `self.settings()`, can the function be a member function on `Ui` (i.e. have a `&self` that's a `Ui`)?

Reply:
We could, though that's inconsistent with `color_setting()` and `pagination_setting()`. Those read `UserSettings` because they're called when initializing a `Ui` (though the progess indicator isn't called that early).

Reply:
Right, I think that's the only reason they're not methods on the `Ui` object; I think they would have been methods on it otherwise.

Reply:
Hm, ok. Then I think this is a good enough reason to just store the pagination setting in the `Ui` struct like you initially suggested.